packages/compass-components/src/components/toggle.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import LeafyGreenToggle from '@leafygreen-ui/toggle';
+
+import { Theme, useTheme } from '../hooks/use-theme';
+
+function Toggle(
+  props: React.ComponentProps<typeof LeafyGreenToggle>
+): ReturnType<typeof LeafyGreenToggle> {
+  const theme = useTheme();
+
+  return <LeafyGreenToggle darkMode={theme?.theme === Theme.Dark} {...props} />;
+}
```

Comment:
Raised this with leafygreen, if that's the only way to set the darkMode for their components, we are at the risk of wrapping every single leafygreen component that we export here which sounds like a giant maintenance headache

Reply:
yes it would be massively better if they'd have a provider we can use, I'd expect LG to also provide themes and guidelines for the palette, which is something we'd have to do on our side otherwise:

```
const darkTheme = {
    theme: Theme.Dark
    colors: {
        accent: palette.green.light1,
        // .. etc
    }
}
```

Reply:
We could do something like this before re-exporting the components on our side:

```
const LGComponentWithTheme<T>(Wrappee: T) {
   const theme = useTheme()
   return (props) => Wrappee(...props, {...darkmode: theme?.theme})
}
```

```
const Toggle = LGComponentWithTheme(Toggle)
export Toggle;
```

Reply:
I mean, yeah, we totally can do something like that, HOC is indeed a good pattern to use here. It's just weird that we have even though we all agree that we want to do as little changes to leafygreen as possible

Reply:
lg team [said](https://mongodb.slack.com/archives/CFFA5BS79/p1643962820401349) this will be addressed when they are eventually working on a dark mode support

Reply:
What should we do for now? I think we probably won't need to do many fixes like this until dark theme/the new provider is released in leafygreen so having one component listening to the theme is probably alright?

Reply:
I would merge this as it is